scripts/envs/open_manipulator/OMReacher_v0.py

```diff
@@ -0,0 +1,368 @@
+        self.moving_state = ""
+        self.actuator_state = ""
+        self.init_robot_pose()
+        rospy.on_shutdown(self._delete_target_block)
+
+
+    def robot_state_callback(self, msg):
+        self.moving_state = msg.open_manipulator_moving_state # "MOVING" / "STOPPED"
+        self.actuator_state = msg.open_manipulator_actuator_state # "ACTUATOR_ENABLE" / "ACTUATOR_DISABLE"
+
+    def joint_state_callback(self, msg):
+        """Callback function of joint states subscriber.
+        Argument: msg
+        """
+        self.joints_states = msg
+        self.joint_names = self.joints_states.name
+        self.joint_positions = self.joints_states.position
+        self.joint_velocities = self.joints_states.velocity
+        self.joint_efforts = self.joints_states.effort
+        # penalize jerky motion in reward for shaped reward setting.
+        self.squared_sum_vel = np.linalg.norm(np.array(self.joint_velocities))
+
+
+    def kinematics_pose_callback(self, msg):
+        """Callback function of gripper kinematic pose subscriber.
+        Argument: msg
+        """
+        self.kinematics_pose = msg
+        _gripper_position = self.kinematics_pose.pose.position
+        self.gripper_position = [_gripper_position.x, _gripper_position.y, _gripper_position.z]
+        _gripper_orientiation = self.kinematics_pose.pose.orientation
+        self.gripper_orientiation = [_gripper_orientiation.x, _gripper_orientiation.y, _gripper_orientiation.z, _gripper_orientiation.w]
+
+    # get and set function
+    # ----------------------------
+
+    def get_joints_states(self):
+        """Returns current joints states of robot including position, velocity, effort
+        Returns: Float64[] self.joints_position, self.joints_velocity, self.joint_effort
+        """
+        return self.joint_positions, self.joint_velocities, self.joint_efforts
+
+    def get_gripper_pose(self):
+        """Returns gripper end effector position
+        Returns: Pose().position, Pose().orientation
+        """
+        return self.gripper_position, self.gripper_orientiation
+
+    def get_gripper_position(self):
+        """Returns gripper end effector position
+        Returns: Pose().position
+        """
+        return self.gripper_position
+
+    def set_joints_position(self, joints_angles):
+
+        """Move joints using joint position command publishers.
+
+        Argument: joints_position_cmd
+        self.joints_position_cmd = [0.0, 0.0, 0.0, 0.0, 0.0, 0.0]
+        """
+        # rospy.loginfo(joints_angles)
+        self.pub_gripper_position.publish(joints_angles[0])
+        self.pub_gripper_sub_position.publish(joints_angles[1])
+        self.pub_joint1_position.publish(joints_angles[2])
+        self.pub_joint2_position.publish(joints_angles[3])
+        self.pub_joint3_position.publish(joints_angles[4])
+        self.pub_joint4_position.publish(joints_angles[5])
+
+    def step(self, action=np.array([1, 1, 1, 1, 1, 1]), step=0):#overriden function
```

Comment:
should have argument only action
[ref: gym.Env](https://github.com/openai/gym/blob/93c5ca0d3d52afdc1e21905f848f2861ad3bbf80/gym/core.py#L46)